tenant-isolation-config-service-impl/src/main/java/org/hypertrace/tenantisolation/config/service/store/TenantIsolationGroupConfigKey.java

```diff
@@ -0,0 +1,15 @@
+package org.hypertrace.tenantisolation.config.service.store;
+
+import java.util.UUID;
+import lombok.Value;
+import org.hypertrace.core.documentstore.Key;
+
+@Value
+public class TenantIsolationGroupConfigKey implements Key {
+  String groupName;
+
+  @Override
+  public String toString() {
+    return UUID.nameUUIDFromBytes((groupName).getBytes()).toString();
```

Comment:
Why do this on each read rather than assigning an ID? Both requires multiple executions and is just generally more confusing.